frontend/src/pages/Match/Timer.js

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react';
+
+function Timer() {
+  const DURATION = 30;
+  const [counter, setCounter] = useState(DURATION);
+
+  useEffect(() => {
+    counter > 0 && setTimeout(() => setCounter(counter - 1), 1000);
```

Comment:
Good practice for react setState function is that if your next state depends on the previous state, it's better to put it in a function, e.g. in this case, setCounter(prevCounter => prevCounter - 1).

This is to prevent unexpected behaviour for say you want to update the counter state twice.
```
setCounter(counter - 1);
setCounter(counter - 1);
```
By right, it should update the counter by decrease by 2 but it will only decrease by 1. Because these two setCounter happens in the same lifecycle of the component and hence both counter's value is say 3 so the next state for both is 2.

If we do:
```
setCounter(prevCounter => prevCounter - 1);
setCounter(prevCounter => prevCounter - 1);
```

Then the first prevCounter is say 3, and the second prevCounter in the second setCounter is 2, therefore resulting in the final state counter = 1.

Although in this case it results in the same counter state since you use it once for every lifecycle. But it's generally a good practice to use function (if next state depends on previous state) to have a more performant and reliable state update.

Reply:
Ah yes, it's been quite awhile since I last used React. Thanks for pointing this out!